services/core-web/src/components/mine/Permit/PermitConditions.spec.tsx

```diff
@@ -52,6 +54,56 @@ describe("PermitConditions", () => {
         </BrowserRouter>
       </ReduxWrapper>
     );
+
+
     expect(container).toMatchSnapshot();
   });
+
+  it("enables adding a condition category in a modal", async () => {
+    render(
+      <ReduxWrapper initialState={initialState} >
+        <BrowserRouter>
+          <ModalWrapper />
+          <ViewPermit />
+        </BrowserRouter>
+      </ReduxWrapper>
+    );
+
+
+    const addConditionLink = screen.getByText("+ Add Condition Category");
+    expect(addConditionLink).toBeInTheDocument();
+    addConditionLink.click();
+
+    let descriptionInput;
+    await waitFor(() => {
+      descriptionInput = screen.getByRole("combobox", { "name": "description" });
+      expect(descriptionInput).toBeInTheDocument();
+    })
+
+
+    const stepInput = screen.getByRole("textbox", { "name": "step" });
+    expect(stepInput).toBeInTheDocument();
+
+    const AddCategoryButton = screen.getByText("Add Category");
+    expect(AddCategoryButton).toBeInTheDocument();
+
+    AddCategoryButton.click();
+
+    await waitFor(() => { expect(screen.getAllByText("This is a required field")).toHaveLength(2); })
+
+    fireEvent.mouseDown(descriptionInput);
```

Comment:
I don't think it's super important, but the docs recommend using userEvent over fireEvent, as it better recreates the actual actions a user would be taking.

Something like:

const user = userEvent.setup()
await user.click(descriptionInput)

Reply:
Yeah agreed. I just did a pass trying to set up the `user-event` package and hit a couple of issues getting the tests converted over. I might leave it for now as is as to not spend too much time, and keep that in mind for next time.

Reply:
Cool cool.  I think userEvent uses fireEvent under the hood and just also simulates some other behaviours as well, but functionally it shouldn't make too much difference.